bridge/setu/broadcaster/broadcaster.go

```diff
@@ -52,41 +59,35 @@ func (tb *TxBroadcaster) BroadcastToHeimdall(msg sdk.Msg) error {
 	txEncoder := helper.GetTxEncoder(tb.cliCtx.Codec)
 	// chain id
 	chainID := helper.GetGenesisDoc().ChainID
-	// current address
-	address := hmTypes.BytesToHeimdallAddress(helper.GetAddress())
+
 	// fetch from APIs
-	var account authTypes.Account
-	response, err := util.FetchFromAPI(tb.cliCtx, util.GetHeimdallServerEndpoint(fmt.Sprintf(util.AccountDetailsURL, address)))
+	account, err := util.GetAccount(tb.cliCtx)
 	if err != nil {
-		tb.logger.Error("Error fetching account from rest-api", "url", util.GetHeimdallServerEndpoint(fmt.Sprintf(util.AccountDetailsURL, address)))
-		panic("Error connecting to rest-server, please start server before bridge.")
-	}
-
-	// get proposer from response
-	if err := tb.cliCtx.Codec.UnmarshalJSON(response.Result, &account); err != nil && len(response.Result) != 0 {
-		tb.logger.Error("Error unmarshalling account details", "url", util.GetHeimdallServerEndpoint(fmt.Sprintf(util.AccountDetailsURL, address)))
+		tb.logger.Error("Error fetching account from rest-api", "url", util.GetHeimdallServerEndpoint(fmt.Sprintf(util.AccountDetailsURL, helper.GetAddress())))
 		return err
 	}
-
 	// get account number and sequence
 	accNum := account.GetAccountNumber()
-	accSeq := account.GetSequence()
 	txBldr := authTypes.NewTxBuilderFromCLI().
 		WithTxEncoder(txEncoder).
 		WithAccountNumber(accNum).
-		WithSequence(accSeq).
+		WithSequence(tb.lastSeqNo).
 		WithChainID(chainID)
 
 	txResponse, err := helper.BuildAndBroadcastMsgs(tb.cliCtx, txBldr, []sdk.Msg{msg})
 	if err != nil {
 		tb.logger.Error("Error while broadcasting the heimdall transaction", "error", err)
+
+		// update seqNo for safety
+		tb.lastSeqNo = account.GetSequence()
```

Comment:
This should be fetched from the rest server instead of using stale account sequence.

Reply:
Fetching from rest-server at line no 64.

Reply:
But that will have a stale sequence. Let's say if you started bridge, and at that time account had sequence 6. No after 3 transactions, it failed due to unexpected failure. If you are using account.GetSequence() then it will give you 6 not 9.

Reply:
Ah got it. sorry.

Reply:
I am fetching account from rest-server every time this method is called for each transaction at line no 64.